website/content/docs/concepts/node-pools.mdx

```diff
@@ -0,0 +1,366 @@
+---
+layout: docs
+page_title: Node Pools
+description: Learn about the internal architecture of Nomad.
+---
+
+# Node Pools
+
+Node pools are a way to group clients and segment infrastructure into logical
+units that can be targeted by jobs for a strong control over where allocations
+are placed.
+
+Without node pools, allocations for a job can be placed in any eligible client
+in the cluster. Affinities and constraints can help express preferences for
+certain nodes, but they do not easily prevent other jobs from placing
+allocations in a set of nodes.
+
+A node pool can be created using the [`nomad node pool apply`][cli_np_apply]
+command and passing a node pool [specification file][np_spec].
+
+```hcl
+# dev-pool.nomad.hcl
+node_pool "dev" {
+  description = "Nodes for the development environment."
+
+  meta {
+    environment = "dev"
+    owner       = "sre"
+  }
+}
+```
+
+```shell-session
+$ nomad node pool apply dev-pool.nomad.hcl
+Successfully applied node pool "dev"!
+```
+
+Clients can then be added to this node pool by setting the
+[`node_pool`][client_np] attribute in their configuration file, or using the
+equivalent [`-node-pool`][cli_agent_np] command line flag.
+
+```hcl
+client {
+  # ...
+  node_pool = "dev"
+  # ...
+}
+```
+
+To help streamline this process, nodes can create node pools on demand. If a
+client configuration references a node pool that does not exist yet, Nomad
+creates the node pool automatically on client registration.
+
+Jobs can then reference node pools using the [`node_pool`][job_np] attribute.
+
+```hcl
+job "app-dev" {
+  # ...
+  node_pool = "dev"
+  # ...
+}
+```
+
+Similarly to the `namespace` attribute, the node pool must exist beforehand,
+otherwise the job registration results in an error. Only nodes in the given
+node pool are considered for placement. If none are available the deployment
+is kept as pending until a client is added to the node pool.
+
+## Multi-region Clusters
+
+In federated multi-region clusters, node pools are automatically replicated
+from the authoritative region to all non-authoritative regions, and requests to
```

Comment:
Not directly related, but I couldn't find what authoritative regions are, maybe add a little phrase about it on the glossary on the region?

Reply:
Ah good point, I added an entry to the glossary about authoritative and non-authoritative regions. I used the same entry because they're opposites so it was kind of weird to explain one without explaining the other 😅 

https://github.com/hashicorp/nomad/pull/18109/files#diff-81af9e6ca20054c572009d730ea51e6a6563ef34befcc20a398fac0c3a6d68d9R96-R106